### DISCUSSION TITLE
How to setup local Supabase `supabase/migrations` folder to mirror schema of remote db

### DISCUSSION THREAD

**Discussion by apuchitnis** [supabase/supabase#6577] [General]
I'm trying to use the following instructions to create a mirror of the remote Supabase db's schema in my local DB: https://supabase.com/docs/guides/local-development#linking-your-project

When I get to the step `supabase db remote set 'postgresql://postgres:<your_password>@db.<your_project_ref>.supabase.co:5432/postgres'` it fails with the following error:

```
Error: The remote database was applied with migration(s) that cannot be found locally. Try updating the project from version control. Otherwise:
1. Delete rows from supabase_migrations.schema_migrations on the remote database so that it's in sync with the contents of supabase/migrations,
2. Run supabase db remote set again,
3. Run supabase db remote commit.```

However, the local `supabase/migrations` folder is empty.

How can I populate the `supabase/migrations` folder with the right data or scripts so that this error doesn't appear? I've tried searching Google/Supabase docs/GitHub, but couldn't find an answer.

Thanks for the help!

**Comment by hanifaudah** [comment]
It's in the docs
https://supabase.com/docs/guides/cli/local-development#deploy-your-project

```
supabase link --project-ref <project-id>
# You can get <project-id> from your project’s dashboard URL: https://app.supabase.com/project/<project-id>

supabase db remote commit
# Capture any changes that you have made to your database before setting up the CLI
```
`project-id` can be found in the settings page of your remote project

After that use this command to finalize the migration
```
supabase db push
```

**Reply by chriscasper** [reply]
That worked. Thank you!

**Comment by juanvilladev** [comment]
Delete your local migrations folder, mine was empty too and I had the same error message. I simply deleted the folder and tried again. It worked.

**Comment by clemwo** [comment]
The solutions in the thread unfortunately don't work for me. Quoting my other reply [here](https://github.com/orgs/supabase/discussions/6366#discussioncomment-6531284)


I'm trying to do the same and get an error.
The general setup is like this:
My colleagues invited me into an already existing supabase project as a developer.
We then created a small basic expo application we want to connect to supabase remote which already works.
Now I'm trying to setup local development with the following steps:
1.   `npx supabase login`
2.  `npx supabase init`
3.  `npx supabase link --project <PROJECT_ID>`
Here I get the error 
> Local config differs from linked project. Try updating supabase/config.toml
and a list of values in the config.toml that apparently aren't in sync with the remote database.
As a next step I tried to pull the database schema from remote to local with
4. `npx supabase db remote commit` and I get the error:
> ERROR: must be owner of event trigger pgsodium_trg_mask_update (SQLSTATE 42501)

Does anyone have a clue how to pull the remote database to local?  
My next step would be to try to set up an empty new supabase project and see if that works. 

Anyway, help is greatly appreciated and I will keep you up to date if I dont forget

**Comment by danvoyce** [comment]
In case anyone like me lands here rather than the docs: https://supabase.com/docs/reference/cli/supabase-db-pull
